### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by andrew17752985**
https://andrew17752985.github.io/react_list-of-goods-js/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
...
+    setGoods(sortedGoods);
+    setAscending(true);
+    setSortAlphabeticallyLight(false);
+    setSortByLengthLight(true);
+    setReverseLight(true);
+    setResetLight(true);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
```
Comment (volodymyr-soltys97) [inline src/App.jsx:39]: You are sorting goods directly which can lead to unstable sorting when items have the same length. Consider using a stable sorting algorithm or adding additional conditions.

File: src/App.jsx [hunk 2]
```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
...
+          type="button"
+          className={`button is-danger ${resetLight ? '' : 'is-light'}`}
+          onClick={resetOrder}
+        >
+          Reset
+        </button>
+      </div>
+
+      <ul>
+        {goods.map((good, index) => (
```
Comment (volodymyr-soltys97) [inline src/App.jsx:106]: Using index as a key is not recommended. If the order of items changes, React will not be able to identify which items have changed, been added, or been removed. Consider using unique string as a key.

File: src/App.jsx [hunk 3]
```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
...
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(initialGoods);
+  const [sortAlphabeticallyLight, setSortAlphabeticallyLight] = useState(true);
```
Comment (volodymyr-soltys97) [inline src/App.jsx:21]: You have multiple states for managing button appearance. Consider refactoring to use a single state object with properties for each button.

File: src/App.jsx [hunk 4]
```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
...
+  const [sortByLengthLight, setSortByLengthLight] = useState(true);
+  const [reverseLight, setReverseLight] = useState(true);
+  const [resetLight, setResetLight] = useState(true);
+  const [ascending, setAscending] = useState(true);
+
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
```
Comment (volodymyr-soltys97) [inline src/App.jsx:28]: You are sorting goods directly which can lead to unstable sorting when items are equal. Consider using a stable sorting algorithm or adding additional conditions.

File: src/App.jsx [hunk 5]
```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
...
+    setGoods(sortedGoods);
+    setAscending(true);
+    setSortAlphabeticallyLight(true);
+    setSortByLengthLight(false);
+    setReverseLight(true);
+    setResetLight(true);
+  };
+
+  const reverseOrder = () => {
+    const reversedGoods = [...goods].reverse();
```
Comment (volodymyr-soltys97) [inline src/App.jsx:50]: Reversing an array can lead to unexpected behavior if you're not careful. When you click 'reverse' twice, it won't necessarily return to the original order. You might want to design a better solution.

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

File: src/App.jsx [hunk 6]
```diff
@@ -14,45 +16,105 @@ export const goodsFromServer = [
...
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(initialGoods);
+  const [buttonStates, setButtonStates] = useState({
```
Comment (anastasiiavorobiova) [inline src/App.jsx:20]: There is no need to use such a complex state here. Consider using 4 states - sort type (blank string, alphabet, length), order, visible, and initial goods. If changing sort type or order -> change state of visible todos occurs.

File: src/App.jsx [hunk 7]
```diff
@@ -14,45 +16,105 @@ export const goodsFromServer = [
...
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          className={`button is-danger ${buttonStates.reset ? '' : 'is-light'}`}
+          onClick={resetOrder}
+        >
+          Reset
```
Comment (anastasiiavorobiova) [inline src/App.jsx:118]: There is no need to render the reverse button in all cases. Consider rendering it if sort or order is applied

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Added comments to fix

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done: 
Screenshot 2023-08-24 at 18 28 24
Consider adding the ability to reverse unsorted list

**Review by nazarmatsevych — APPROVED** [review]

**Pull request closed without merging**